crates/sui/src/upgrade_compatibility.rs

```diff
@@ -813,6 +821,47 @@ fn module_compatibility_error_diag(
     Ok(diags)
 }
 
+fn missing_module_diag(
+    module_name: &Identifier,
+    package_path: &PathBuf,
+) -> Result<Diagnostics, Error> {
+    let mut diags = Diagnostics::new();
+
+    // read Move.toml to get the hash and first line start and end
+    let toml_path = package_path.join("Move.toml");
+    let toml_str = fs::read_to_string(&toml_path).context("Unable to read Move.toml")?;
+    let hash = FileHash::new(&toml_str);
+    let chars: Vec<char> = toml_str.chars().collect();
+
+    let start = toml_str
+        .chars()
+        .enumerate()
+        .find(|(_, c)| !c.is_whitespace())
+        .map(|(i, _)| i)
+        .unwrap_or_default();
+
+    let end = chars[start..]
+        .iter()
+        .enumerate()
+        .find(|(_, c)| c.is_whitespace())
+        .map(|(i, _)| i)
+        .unwrap_or_default();
+
+    let loc = Loc::new(hash, start as ByteIndex, end as ByteIndex);
```

Comment:
Using character offsets will not always map to byte offsets, unfortunately. In Rust, a `char` corresponds to a single Unicode Scalar Value, which may be encoded in between 1 to 4 UTF8 bytes in the `&str`, so you are better off working with the `&str` directly, and calling [`find`](https://doc.rust-lang.org/std/primitive.str.html#method.find) on it.

In terms of what to look for, I would try and find the `[package]` section and slice out some number of lines after it to have a good chance of finding a name.